tests/platform_tests/counterpoll/test_counterpoll_watermark.py

```diff
@@ -237,14 +258,15 @@ def verify_counterpoll_status(duthost, counterpoll_list, expected):
 
 
 def count_watermark_stats_in_counters_db(duthost):
-    watermark_stats_output = redis_get_keys(duthost, 'COUNTERS_DB', '*{}*'
-                                            .format(CounterpollConstants.WATERMARK.upper()))
-    watermark_stats = {}
-    for watermark_type in WATERMARK_COUNTERS_DB_STATS_TYPE:
-        watermark_stats[watermark_type] = 0
-        for line in watermark_stats_output:
-            if watermark_type in line:
-                watermark_stats[watermark_type] += 1
-    logging.info("watermark_stats {}".format(watermark_stats))
-    for k, v in list(watermark_stats.items()):
-        pytest_assert(v > 0, "watermark_stats {} in COUNTERS_DB: {}, expected > 0".format(k, v))
+    watermark_stats_output = get_keys_on_asics(duthost, 'COUNTERS_DB', '*{}*'
+                                               .format(CounterpollConstants.WATERMARK.upper()))
+    for asic_idx, output in watermark_stats_output.items():
+        watermark_stats = {}
+        for watermark_type in WATERMARK_COUNTERS_DB_STATS_TYPE:
+            watermark_stats[watermark_type] = 0
+            for line in output:
+                if watermark_type in line:
+                    watermark_stats[watermark_type] += 1
```

Comment:
The test case will not detect if one of the asic does not have any watermark stat?

Reply:
it iterates through each asic one at a time - if it finds one where there is missing watermarks itll assert and stop.